api/internal/git/repospec_test.go

```diff
@@ -35,6 +35,8 @@ func TestNewRepoSpecFromUrl_Permute(t *testing.T) {
 		{"git::https://git.example.com/", "https://git.example.com/"},
```

Comment:
I started reading a bit more into this as well. Basing on this [StackOverflow](https://stackoverflow.com/questions/3248779/which-is-the-smarter-git-protocol-ssh-or-gitover-ssh-or-https-protocol) and reading the code in go-getter, I don't believe `git::ssh://` is a valid combination as `git://` is its own standalone protocol. I added `git::git@github.com:`, but could not add `git::git@github.com/` because it does not behave properly.

In the context of username parsing, I think `git::` prefix should continue to only allow `git@` as the user. My PR does not appear to change that based on my testing.

Reply:
Do we still need to support `git::`? I did a git blame on this code and found that we introduced `git::` in [this PR](https://github.com/kubernetes-sigs/kustomize/pull/303) back when kustomize still used go-getter. Since we've removed the go-getter dependency, can we stop handling `git::` as well? 

If we still need to continue to support inputs that we once supported, at least in the same major version, I'd understand why we need to keep this.

Reply:
> I don't believe git::ssh:// is a valid combination as git:// is its own standalone protocol.

While the background is interesting and does suggest it may be conceptually invalid, what matters for us in this case is whether or not it has already been working. Imagine a user is using that prefix in a bunch of Kustomizations--are they able to successfully execute `kustomize build` on them? If they are, it needs to continue to work. Since part of the go-getter readme [here](https://github.com/hashicorp/go-getter#git-git) explicitly gives an example with this combo, I'm guessing it does work and needs to be retained as a result.

> Since we've removed the go-getter dependency, can we stop handling git:: as well?

The only time we could revisit this is is in a major version bump, but even then we want to avoid causing toil for our users without a very compelling reason.

Reply:
Added a regression test in 62c772ab5

Reply:
> Since part of the go-getter readme [here](https://github.com/hashicorp/go-getter#git-git) explicitly gives an example with this combo, I'm guessing it does work and needs to be retained as a result.

Understood. Thanks for the explanation, @KnVerey!